tracing/debug/tripperware.go

```diff
@@ -25,25 +26,39 @@ func Tripperware(opts ...Option) httpwares.Tripperware {
 			}
 			tr := trace.New(operationNameFromUrl(req), req.URL.String())
 			defer tr.Finish()
+
 			tr.LazyPrintf("%v %v HTTP/%d.%d", req.Method, req.URL, req.ProtoMajor, req.ProtoMinor)
-			tr.LazyPrintf("Host: %v", hostFromReq(req))
+			var reqhbuf bytes.Buffer
 			for k, _ := range req.Header {
-				tr.LazyPrintf("%v: %v", k, req.Header.Get(k))
+				fmt.Fprintf(&reqhbuf, "%v: %v\n", k, req.Header.Get(k))
+			}
+			if reqhbuf.Len() > 0 {
+				reqhbuf.Truncate(reqhbuf.Len() - 1)
+				tr.LazyPrintf("%s", &reqhbuf)
 			}
-			tr.LazyPrintf("invoking next chain")
+
 			resp, err := next.RoundTrip(req)
-			tr.LazyPrintf("tags: ")
+
+			var tbuf bytes.Buffer
+			tbuf.WriteString("tags:")
 			for k, v := range http_ctxtags.ExtractInbound(req).Values() {
-				tr.LazyPrintf("%v: %v", k, v)
+				fmt.Fprintf(&tbuf, " %v=%q", k, v)
 			}
+			tr.LazyPrintf("%s", &tbuf)
+
 			if err != nil {
 				tr.LazyPrintf("Error on response: %v", err)
 				tr.SetError()
 			} else {
-				tr.LazyPrintf("HTTP/%d.%d %d %s", resp.ProtoMajor, resp.ProtoMinor, resp.StatusCode, resp.StatusCode)
-				tr.LazyPrintf("Content-Length:  %d", resp.Status, resp.ContentLength)
+				tr.LazyPrintf("HTTP/%d.%d %d %s", resp.ProtoMajor, resp.ProtoMinor, resp.StatusCode, resp.Status)
+				tr.LazyPrintf("Content-Length: %d", resp.ContentLength)
+				var reshbuf bytes.Buffer
 				for k, _ := range resp.Header {
-					tr.LazyPrintf("%v: %v", k, resp.Header.Get(k))
+					fmt.Fprintf(&reshbuf, "%v: %v\n", k, resp.Header.Get(k))
```

Comment:
You could put these 3 places you are constructing buff from header to single function.. but not a blocker